app/components/flash-display.js

```diff
@@ -1,15 +1,23 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { alias } from '@ember/object/computed';
 
 export default Component.extend({
   flashes: service(),
 
-  classNames: ['flash'],
+  classNameBindings: ['type'],
   tagName: 'ul',
 
   messages: alias('flashes.messages'),
 
+  type: computed('messages.@each.type', function () {
+    const classes = this.messages.map(message => message.type).reduce(
+      (unique, item) => (unique.includes(item) ? unique : [...unique, item]),
+      []).join(' ');
```

Comment:
You can use Ember's [`uniqBy`](https://api.emberjs.com/ember/3.12/classes/EmberArray/methods/uniqBy?anchor=uniqBy) and [`mapBy`](https://api.emberjs.com/ember/3.12/classes/EmberArray/methods/mapBy?anchor=mapBy) array extensions to simplify this logic:
```suggestion
    const classes = this.messages.uniqBy('type').mapBy('type').join(' ');
```